pkg/config/config.go

```diff
@@ -168,8 +168,14 @@ func (fc *FulcioConfig) GetVerifier(issuerURL string, opts ...InsecureOIDCConfig
 	// If this issuer hasn't been recently used, or we have special config options, then create a new verifier
 	// and add it to the LRU cache.
 
-	ctx, cancel := context.WithTimeout(context.Background(), defaultOIDCDiscoveryTimeout)
+	parentContext, cancel := context.WithTimeout(context.Background(), defaultOIDCDiscoveryTimeout)
 	defer cancel()
+
+	// go-oidc use DefaultClient and it does not take into account our custom config of http.DefaultTransport
+	// ClientContext create a new Context that carries our http client.
+
+	oidcClient := &http.Client{}
```

Comment:
I might be misunderstanding, but isn't this equivalent to `http.DefaultClient`?

Reply:
https://cs.opensource.google/go/go/+/refs/tags/go1.21.4:src/net/http/client.go;drc=97df9c0b051b6c06bff7ec7e61522e7bbef40c91;l=30
```
// A Client is an HTTP client. Its zero value (DefaultClient) is a
// usable client that uses DefaultTransport.
```

Reply:
You are both correct, my problem was elsewhere, I backported the previous patch to my 1.4.0 code and I thought it solved issue but I also imported this change:

```go
diff --git a/pkg/config/config.go b/pkg/config/config.go
index 45f0c17..9068bd0 100644
--- a/pkg/config/config.go
+++ b/pkg/config/config.go
@@ -229,10 +229,11 @@ func (fc *FulcioConfig) prepare() error {
                defer cancel()
                provider, err := oidc.NewProvider(ctx, iss.IssuerURL)
                if err != nil {
-                       return fmt.Errorf("provider %s: %w", iss.IssuerURL, err)
+                       log.Logger.Errorf("error creating provider for issuer URL %q: %v", iss.IssuerURL, err)
+               } else {
+                       cfg := &oidc.Config{ClientID: iss.ClientID}
+                       fc.verifiers[iss.IssuerURL] = []*verifierWithConfig{{provider.Verifier(cfg), cfg}}
                }
-               cfg := &oidc.Config{ClientID: iss.ClientID}
-               fc.verifiers[iss.IssuerURL] = []*verifierWithConfig{{provider.Verifier(cfg), cfg}}
        }

        cache, err := lru.New2Q(100 /* size */)
```

Which made the pod running even though no valid OIDC was present.